resolwe_bio/processes/samtools/samtools_coverage.py

```diff
@@ -0,0 +1,162 @@
+                "base pairs",
+            )
+            min_mq = IntegerField(
+                label="Minimum mapping quality",
+                required=False,
+                description="Minimum mapping quality for an alignment to be used.",
+            )
+            min_bq = IntegerField(
+                label="Minimum base quality",
+                required=False,
+                description="Minimum base quality for a base to be considered.",
+            )
+            excl_flags = StringField(
+                label="Filter flags",
+                required=False,
+                description="Filter flags: skip reads with mask bits set "
+                "[UNMAP,SECONDARY,QCFAIL,DUP]",
+            )
+            depth = IntegerField(
+                label="Maximum allowed coverage depth",
+                required=False,
+                description="If 0, depth is set to the maximum integer value "
+                "effectively removing any depth limit.",
+            )
+            no_header = BooleanField(
+                label="No header",
+                default=False,
+                description="Don't print a header in tabular mode.",
+            )
+
+        advanced = GroupField(AdvancedOptions, label="Advanced options")
+
+    class Output:
+        """Output fields for SamtoolsCoverage."""
+
+        table = FileField(label="Output coverage table")
+        build = StringField(label="Build")
+        species = StringField(label="Species")
+
+    def run(self, inputs, outputs):
+        """Run the analysis."""
+
+        output_name = "samtools_coverage.tsv"
+
+        num_of_bams = len(inputs.bam)
+        if num_of_bams > 1:
+            for i in range(1, num_of_bams):
+                if inputs.bam[0].output.species != inputs.bam[i].output.species:
+                    self.error("Not all BAM files are from the same species.")
+
+                if inputs.bam[0].output.build != inputs.bam[i].output.build:
+                    self.error("Not all BAM files have the same genome build.")
+
+        input_options = [
+            "-o",
+            output_name,
+        ]
+
+        if inputs.region:
+            input_options.extend(["--region", inputs.region])
+
+        if inputs.advanced.min_read_length:
+            input_options.extend(["--min-read-len", inputs.advanced.min_read_length])
+
+        if inputs.advanced.min_mq:
+            input_options.extend(["--min-MQ", inputs.advanced.min_mq])
+
+        if inputs.advanced.min_bq:
+            input_options.extend(["--min-BQ", inputs.advanced.min_bq])
+
+        if inputs.advanced.excl_flags:
+            input_options.extend(["--excl-flags", inputs.advanced.excl_flags])
+
+        if inputs.advanced.depth:
+            input_options.extend(["--depth", inputs.advanced.depth])
+
+        if inputs.advanced.no_header:
+            input_options.append("--no-header")
+
+        bam_files = [inputs.bam[i].output.bam.path for i in range(num_of_bams)]
```

Comment:
Range could be avoided here as well `[bam.output.bam.path for bam in inputs.bam]`